src/status_im/subs/communities.cljs

```diff
@@ -71,51 +73,82 @@
-     [profile members visibility-status-updates my-status-update]))
- (fn [[profile members visibility-status-updates my-status-update] _]
-   (let [online? (fn [public-key]
-                   (let [{visibility-status-type :status-type}
-                         (if (or (string/blank? (:public-key profile))
-                                 (= (:public-key profile) public-key))
-                           my-status-update
-                           (get visibility-status-updates public-key))]
-                     (subs.utils/online? visibility-status-type)))
-         names   (keys->names (keys members) profile)]
-     (->> members
-          (sort-members-by-name names true)
-          keys
-          (group-by online?)
-          (map (fn [[k v]]
-                 {:title (if k (i18n/label :t/online) (i18n/label :t/offline))
-                  :data  v}))))))
+   [(re-frame/subscribe [:profile/public-key])
+    (re-frame/subscribe [:communities/community-chat-members community-id chat-id])
+    (re-frame/subscribe [:visibility-status-updates])
+    (re-frame/subscribe [:multiaccount/current-user-visibility-status])])
+ (fn [[profile-pub-key members-js visibility-status-updates my-status-update] [_ _ _ visibility-status]]
+   (let [members-keys (js-keys members-js)
+         online?      (fn [public-key]
+                        (let [{visibility-status-type :status-type}
+                              (if (or (string/blank? profile-pub-key)
+                                      (= profile-pub-key public-key))
+                                my-status-update
+                                (get visibility-status-updates public-key))]
+                          (subs.utils/online? visibility-status-type)))]
+     (filter (if (= :online visibility-status)
+               online?
+               (complement online?))
+             members-keys))))
+
+(defn- names-by-key
+  [contacts profile public-keys]
+  (let [names (reduce (fn [acc k]
+                        (if-let [contact (get contacts k)]
+                          (assoc acc k (contact.utils/contact-two-names contact profile))
+                          acc))
+                      {}
+                      public-keys)]
+    (assoc names
+           (:public-key profile)
+           [(profile.utils/displayed-name profile) nil])))
+
+;; This is a potentially expensive subscription because we don't control how
+;; many members and contacts exist in the app-db. Future improvements include
+;; removing members from the payload and paginating them from status-go.
+(re-frame/reg-sub :communities/chat-members-sorted
+ (fn [[_ community-id chat-id visibility-status]]
+   [(re-frame/subscribe [:profile/profile])
+    (re-frame/subscribe [:contacts/contacts-raw])
+    (re-frame/subscribe [:communities/chat-members community-id chat-id visibility-status])])
+ (fn [[profile contacts ^js members-keys]]
+   (sort-members-by-name (names-by-key contacts profile members-keys)
```

Comment:
If we are consuming this large dataset from a flat-list, we could provide indexes so that the paging is done on our side instead of Status-go.

Not an immediate change, just saying we could process it lazily according to the user's scrolling.

Reply:
The goal, or at least my personal goal, is to eventually remove members from the community payload coming from status-go. There can be just way too many members in real life, and this won't scale well for mobile. The data should be paginated in status-go, like what we do with notifications for example. Also sorting should be handled better directly by status-go.

So all in all @ulisesmac, I think this is a decent solution for now, but I would prefer to see most of this code disappear from the client.